cli/src/commands/generate/generate.ts

```diff
@@ -132,22 +135,46 @@ function instantiateRelationships(pattern: any): any {
     return outputRelationships;
 }
 
+function instantiateAdditionalTopLevelProperties(pattern: any): any {
+    const properties = pattern?.properties;
+    if (!properties) {
+        console.error('Warning: pattern has no properties defined.');
```

Comment:
this should be logger instead of console

Reply:
Fixed